system/lib/wasm_worker/library_wasm_worker.c

```diff
@@ -22,42 +24,52 @@ void __wasm_init_tls(void *memory);
 __attribute__((constructor(48)))
 static void emscripten_wasm_worker_main_thread_initialize() {
 	uintptr_t* sbrk_ptr = emscripten_get_sbrk_ptr();
-	assert((*sbrk_ptr & 15) == 0);
-	assert(__builtin_wasm_tls_align() <= 16);
+	*sbrk_ptr = ROUND_UP(*sbrk_ptr, __builtin_wasm_tls_align());
```

Comment:
Given that `__builtin_wasm_tls_align()` is a compile time construct, I wonder if there would be a way to make this a no-op when `__builtin_wasm_tls_align() <= 16`, which I'd say is 99% of all use cases?

Reply:
Maybe a

```cpp
assert((*sbrk_ptr & 15) == 0);
if (__builtin_wasm_tls_align() > 16) {
  *sbrk_ptr = ROUND_UP(*sbrk_ptr, __builtin_wasm_tls_align());
}
assert((*sbrk_ptr & (__builtin_wasm_tls_align()-1)) == 0);
```
might do the trick, to help wasm-opt optimize here? (I'd keep the asserts like above so in case of issues, we'd know how sbrk ptr and `__builtin_wasm_tls_align()` relate to each other)